### DISCUSSION TITLE
Git Repo vs Helm Repo

### DISCUSSION THREAD

**Discussion by mindrunner** [emqx/emqx#7811] [General]
The helm chart was updated 10 days ago (Prometheus Monitoring). However, the latest chart in Helm Repo still lacks that feature? 
Shouldn't that be released any soon? :)

**Comment by zmstone** [comment]
@Rory-Z we may need to automate this.

**Reply by mindrunner** [reply]
Sounds like a good idea :) 
Until then, can we have this feature soon? :)

**Comment by Rory-Z** [comment]
For some historical reason the emqx helm chart is released in sync with emqx and the version of the emqx helm chart is also in sync with emqx, this is what bothers me, maybe we should keep the version of the emqx helm chart separate from the version of emqx ? 
But the problem is that the emqx helm chart has been released to 4.4.3, so if the emqx helm chart is released separately, what will happen to its version number?
@zmstone

**Reply by advanisunny3** [reply]
Apologies for speaking out of turn, but sounds like a good idea to have separate versions of helm and emqx. 
I have seen other open source software also following same practice, for example Airflow helm chart version is 1.5.0 but application version(airflow) is 2.2.3

**Reply by zmstone** [reply]
> For some historical reason the emqx helm chart is released in sync with emqx and the version of the emqx helm chart is also in sync with emqx, this is what bothers me, maybe we should keep the version of the emqx helm chart separate from the version of emqx ? But the problem is that the emqx helm chart has been released to 4.4.3, so if the emqx helm chart is released separately, what will happen to its version number? @zmstone

I think it's a good idea to be flexible, but I do wish to have very intuitive mapping from EMQX version to chart version(vice versa)
So that as a chart user, I do not have to search some other place to find out which exact version I should pick or currently using.

If chart version can be flexible, it maybe can be `0.1.0-emqx-v4.3.14` ?

**Reply by zmstone** [reply]
> Apologies for speaking out of turn, but sounds like a good idea to have separate versions of helm and emqx. I have seen other open source software also following same practice, for example Airflow helm chart version is 1.5.0 but application version(airflow) is 2.2.3

Thank you for the tip.

**Reply by zmstone** [reply]
or does it make more sense to have a helm repo for each emqx minor release? e.g. `emqx-4.3` and the chart version can be semver ?

**Reply by mindrunner** [reply]
Usually chart version and app version is decoupled. Even in this case, it is possible to use 4.4.3 chart with 4.4.1 emqx for example by overriding the docker image tag.

**Reply by advanisunny3** [reply]
Yep, plus the dependency of helm chart will be with Kubernetes version. We might need to update helm chart due to K8 API changes and still keep same emqx version.
We can see the application and chart version separately too, if we add appVersion to chart.yml
Here's example from artifacthub

![image](https://user-images.githubusercontent.com/6717653/165930412-8fbed0d8-3f50-4f10-8a05-dcbb1f8ff137.png)

**Reply by Rory-Z** [reply]
Yes, I am in favour of chart version and app version is decoupled, but the problem is that the emqx helm chart version has been released to 4.4.3, just like the emqx version, so how should we name the new chart version to maintain the semantics and not confuse others?

**Reply by Rory-Z** [reply]
Apologies for the delay, I've just returned from a holiday

**Reply by advanisunny3** [reply]
Sorry, I have also been on holiday and missed to reply.
How about we start the practice from v5 release onwards? Initially, we can keep chart and app version to 5.0.0 and then manage chart versions independent of appVersion. This way it would provide members enough time to adopt new versioning scheme of helm.

I think it would also be good to define what exactly contributes as chart version change.

**Reply by mindrunner** [reply]
How about releasing the current state of the chart as `4.4.3_r1` or similar? Would be super nice to have to have the monitoring feature asap. :)

**Reply by Rory-Z** [reply]
Yes, maybe we can set chart.version to `4.4.3-1`, check again, is that all code we need ?: https://github.com/emqx/emqx/pull/7714

**Reply by mindrunner** [reply]
Sorry for the late reply, I thought I already answered this. I am mainly interested in this change: https://github.com/emqx/emqx/pull/7643

**Reply by Rory-Z** [reply]
Cool, I create a [new PR](https://github.com/emqx/emqx/pull/8060) for EMQX 4.3, and the will be sync to EMQX 4.4, afterwards we will release new version

**Reply by mindrunner** [reply]
Not sure if I understand. It seem it got merged, but I still cannot see an updated chart version.

**Reply by Rory-Z** [reply]
Sorry for the late， the EMQX will plan release 4.4.4 for tomorrow，so this change will be include

**Reply by mindrunner** [reply]
Sweet, thanks! Looking forward to that :)

**Comment by agronholm** [comment]
Sorry to bug you, but as the latest released chart version doesn't support SSL, it would be really nice to get v5.0.6 of the chart (which does have that) released to the helm repo. Thank you in advance!

**Reply by Links2004** [reply]
latest helm for v5 is 5.0.3 looks like releasing the helm chart is not part of the release todo.
and I too want to use SSL.

**Reply by mindrunner** [reply]
@Rory-Z @zmstone can we have a talk about this please? I am happy to help with the release process if you need support. But this is kinda important. Otherwise it doesn't make sense to maintain the chart in repo. I am currently using a private fork to have a smooth transition to newer versions...

**Reply by Rory-Z** [reply]
Hi @mindrunner I'm sorry, we missed the update for helm chart, the helm chart will be update for next version, please check this PR: https://github.com/emqx/emqx/pull/8921

Also I recommend trying EMQX Operator, please check this: https://github.com/emqx/emqx-operator/blob/2.0.0/docs/en_US/getting-started/getting-started.md

@mindrunner Thank you for your code contribution to the EMQX Helm Chart, very great, we look forward to your use of the EMQX Operator!

**Reply by mindrunner** [reply]
Just saw the operator yesterday. excited to move over. Is there some sort of migration guide for existing installations?

**Reply by Rory-Z** [reply]
Sorry, we don't have documentation for converting EMQX Helm Chart to EMQX Custom Resource at the moment, I think it's two concepts and it's difficult to convert it

**Reply by agronholm** [reply]
I've built and published the v5.0.7 chart here if you want to use it: https://agronholm.github.io/emqx/

**Reply by agronholm** [reply]
I found the operator difficult to use because I could not find documentation for the custom resource used for configuration.

**Reply by Rory-Z** [reply]
Hi @agronholm please check this: https://github.com/emqx/emqx-operator/blob/2.0.0/docs/en_US/getting-started/getting-started.md
If you have any problem for EMQX Operator, welcome to create issue for https://github.com/emqx/emqx-operator/issues
